src/main/java/uk/gov/companieshouse/api/accounts/controller/CurrentPeriodController.java

```diff
@@ -100,6 +105,55 @@ public ResponseEntity create(@RequestBody @Valid CurrentPeriod currentPeriod,
         return responseEntity;
     }
 
+    @PutMapping
+    public ResponseEntity update(@RequestBody @Valid CurrentPeriod currentPeriod,
+        BindingResult bindingResult, @PathVariable("companyAccountId") String companyAccountId,
+        HttpServletRequest request) {
+
+        SmallFull smallFull = (SmallFull) request.getAttribute(AttributeName.SMALLFULL.getValue());
+        if (smallFull.getLinks().get(SmallFullLinkType.CURRENT_PERIOD.getLink()) == null) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+
+        Errors errors = new Errors();
+
+        if (bindingResult.hasErrors()) {
+            errors = errorMapper.mapBindingResultErrorsToErrorModel(bindingResult, errors);
+        }
+
+        currentPeriodValidator.validateCurrentPeriod(currentPeriod, errors);
+        if (errors.hasErrors()) {
+            LOGGER.error(
+                "Current period validation failure");
+            logValidationFailureError(getRequestId(request), errors);
+            return new ResponseEntity<>(errors, HttpStatus.BAD_REQUEST);
+
+        }
+
+        Transaction transaction = (Transaction) request
+            .getAttribute(AttributeName.TRANSACTION.getValue());
+
+        String requestId = request.getHeader(REQUEST_ID);
+
+        ResponseEntity responseEntity;
+
+        try {
+            ResponseObject<CurrentPeriod> responseObject = currentPeriodService
+                .update(currentPeriod, transaction, companyAccountId, requestId);
+            responseEntity = apiResponseMapper
+                .map(responseObject.getStatus(), responseObject.getData(),
```

Comment:
the second argument should be null here to avoid returning a body - the spec for the PUT does not contain a response body